### ISSUE TITLE
It seems pyrakoon also raises ArakoonSockRecvError these days

### ISSUE THREAD

**Issue by khenderick** [openvstorage/framework#942]
It seems pyrakoon also raises `ArakoonSockRecvError` these days. We should also incorporate the retry for that error.

**Comment by khenderick** [comment]
Logging of a few threads/processes running at the same time, all suffering from some issue.

```
2016-09-22 09:31:31 97400 +0200 - ovs100 - 26213/140079970371392 - extensions/pyrakoon_client - 274 - DEBUG - Error during arakoon call get, retry
2016-09-22 09:31:32 97600 +0200 - ovs100 - 26213/140079970371392 - extensions/pyrakoon_client - 275 - WARNING - Arakoon call get took 64.1s


2016-09-22 09:31:31 97400 +0200 - ovs100 - 26209/140079970371392 - arakoon_client/pyrakoon - 269 - ERROR - timed out: Error while reading socket
Traceback (most recent call last):
  File "/opt/OpenvStorage/ovs/extensions/db/arakoon/pyrakoon/pyrakoon/compat.py", line 1342, in read
    data = self._socket.recv(bytes_remaining)
timeout: timed out
2016-09-22 09:31:31 97500 +0200 - ovs100 - 26209/140079970371392 - extensions/pyrakoon_client - 270 - ERROR - Error during get. Process 26209, thread 140079970371392, clientid 5155680


2016-09-22 09:31:31 97600 +0200 - ovs100 - 17887/140079970371392 - arakoon_client/pyrakoon - 44 - ERROR - timed out: Error while reading socket
Traceback (most recent call last):
  File "/opt/OpenvStorage/ovs/extensions/db/arakoon/pyrakoon/pyrakoon/compat.py", line 1342, in read
    data = self._socket.recv(bytes_remaining)
timeout: timed out
2016-09-22 09:31:31 97600 +0200 - ovs100 - 17887/140079970371392 - extensions/pyrakoon_client - 45 - ERROR - Error during get. Process 17887, thread 140079970371392, clientid 5155680
2016-09-22 09:31:31 97700 +0200 - ovs100 - 17887/140079970371392 - celery/celery.redirected - 47 - WARNING - 2016-09-22 09:31:31 97700 +0200 - ovs100 - 17887/140079970371392 - extensions/volatile mutex - 46 - WARNING - A lock on ovs_lock_voldrv_event_disk_f8b0edd5-5d09-4f31-b92d-d6cbe46723ac was kept for 63.5493280888 sec


2016-09-22 09:31:31 97700 +0200 - ovs100 - 26188/140079970371392 - extensions/pyrakoon_client - 250 - DEBUG - Error during arakoon call get, retry
2016-09-22 09:31:32 97900 +0200 - ovs100 - 26188/140079970371392 - extensions/pyrakoon_client - 251 - WARNING - Arakoon call get took 61.06s


2016-09-22 09:31:31 98200 +0200 - ovs100 - 17866/140079970371392 - celery/celery.worker.job - 387 - ERROR - Task ovs.vdisk.resize_from_voldrv[24d4c2fc-4419-478c-a2b7-5589e9fc3c4e] raised unexpected: ArakoonSockRecvError('Error while receiving data from socket',)
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/celery/app/trace.py", line 240, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/celery/app/trace.py", line 438, in __protected_call__
    return self.run(*args, **kwargs)
  File "/opt/OpenvStorage/ovs/lib/helpers/decorators.py", line 69, in new_function
    return function(*args, **kwargs)
  File "/opt/OpenvStorage/ovs/lib/vdisk.py", line 178, in resize_from_voldrv
    VDiskController._set_vdisk_metadata_pagecache_size(vdisk)
  File "/opt/OpenvStorage/ovs/lib/vdisk.py", line 1154, in _set_vdisk_metadata_pagecache_size
    storagedriver_config.load()
  File "/opt/OpenvStorage/ovs/extensions/storageserver/storagedriver.py", line 257, in load
    self.configuration = json.loads(Configuration.get(self.key, raw=True))
  File "/opt/OpenvStorage/ovs/extensions/generic/configuration.py", line 112, in get
    data = Configuration._get(key_entries[0], raw)
  File "/opt/OpenvStorage/ovs/extensions/generic/configuration.py", line 362, in _get
    key=key)
  File "/opt/OpenvStorage/ovs/extensions/generic/configuration.py", line 404, in _passthrough
    return getattr(ArakoonConfiguration, method)(*args, **kwargs)
  File "/opt/OpenvStorage/ovs/extensions/db/arakoon/configuration.py", line 119, in get
    return client.get(key)
  File "/opt/OpenvStorage/ovs/extensions/db/arakoon/pyrakoon/client.py", line 44, in new_function
    return f(self, *args, **kw)
  File "/opt/OpenvStorage/ovs/extensions/db/arakoon/pyrakoon/client.py", line 76, in get
    return PyrakoonClient._try(self._identifier, self._client.get, key, consistency)
  File "/opt/OpenvStorage/ovs/extensions/db/arakoon/pyrakoon/client.py", line 207, in _try
    return_value = method(*args, **kwargs)
  File "<update_argspec>", line 5, in get
  File "/opt/OpenvStorage/ovs/extensions/db/arakoon/pyrakoon/pyrakoon/compat.py", line 145, in wrapped
    return fun(*args, **kwargs)
  File "/opt/OpenvStorage/ovs/extensions/db/arakoon/pyrakoon/pyrakoon/compat.py", line 128, in wrapped
    return fun(*new_args)
  File "/opt/OpenvStorage/ovs/extensions/db/arakoon/pyrakoon/pyrakoon/compat.py", line 246, in get
    return self._client.get(key,consistency = consistency_)
  File "<update_argspec>", line 5, in get
  File "/opt/OpenvStorage/ovs/extensions/db/arakoon/pyrakoon/pyrakoon/client/utils.py", line 99, in wrapped
    return self._process(message) #pylint: disable=W0212
  File "/opt/OpenvStorage/ovs/extensions/db/arakoon/pyrakoon/pyrakoon/compat.py", line 1103, in _process
    connection.read)
  File "/opt/OpenvStorage/ovs/extensions/db/arakoon/pyrakoon/pyrakoon/utils.py", line 342, in read_blocking
    value = read_fun(request.count)
  File "/opt/OpenvStorage/ovs/extensions/db/arakoon/pyrakoon/pyrakoon/compat.py", line 1347, in read
    raise ArakoonSockRecvError
ArakoonSockRecvError: Error while receiving data from socket
2016-09-22 09:31:32 00800 +0200 - ovs100 - 17866/140079970371392 - celery/celery.worker.job - 388 - ERROR - Task ovs.vdisk.resize_from_voldrv[4bbf0ff3-f687-4072-8b50-11c4fd176f95] raised unexpected: ArakoonSockRecvError('Error while receiving data from socket',)
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/celery/app/trace.py", line 240, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/celery/app/trace.py", line 438, in __protected_call__
    return self.run(*args, **kwargs)
  File "/opt/OpenvStorage/ovs/lib/helpers/decorators.py", line 56, in new_function
    metadata = {'storagedriver': StorageDriverList.get_by_storagedriver_id(kwargs['storagedriver_id']).guid}
  File "/opt/OpenvStorage/ovs/dal/lists/storagedriverlist.py", line 43, in get_by_storagedriver_id
    'items': [('storagedriver_id', DataList.operator.EQUALS, storagedriver_id)]})
  File "/opt/OpenvStorage/ovs/dal/datalist.py", line 82, in __init__
    self._volatile = VolatileFactory.get_client()
  File "/opt/OpenvStorage/ovs/extensions/storage/volatilefactory.py", line 38, in get_client
    client_type = Configuration.get('/ovs/framework/stores|volatile')
  File "/opt/OpenvStorage/ovs/extensions/generic/configuration.py", line 112, in get
    data = Configuration._get(key_entries[0], raw)
  File "/opt/OpenvStorage/ovs/extensions/generic/configuration.py", line 362, in _get
    key=key)
  File "/opt/OpenvStorage/ovs/extensions/generic/configuration.py", line 404, in _passthrough
    return getattr(ArakoonConfiguration, method)(*args, **kwargs)
  File "/opt/OpenvStorage/ovs/extensions/db/arakoon/configuration.py", line 119, in get
    return client.get(key)
  File "/opt/OpenvStorage/ovs/extensions/db/arakoon/pyrakoon/client.py", line 44, in new_function
    return f(self, *args, **kw)
  File "/opt/OpenvStorage/ovs/extensions/db/arakoon/pyrakoon/client.py", line 76, in get
    return PyrakoonClient._try(self._identifier, self._client.get, key, consistency)
  File "/opt/OpenvStorage/ovs/extensions/db/arakoon/pyrakoon/client.py", line 207, in _try
    return_value = method(*args, **kwargs)
  File "<update_argspec>", line 5, in get
  File "/opt/OpenvStorage/ovs/extensions/db/arakoon/pyrakoon/pyrakoon/compat.py", line 145, in wrapped
    return fun(*args, **kwargs)
  File "/opt/OpenvStorage/ovs/extensions/db/arakoon/pyrakoon/pyrakoon/compat.py", line 128, in wrapped
    return fun(*new_args)
  File "/opt/OpenvStorage/ovs/extensions/db/arakoon/pyrakoon/pyrakoon/compat.py", line 246, in get
    return self._client.get(key,consistency = consistency_)
  File "<update_argspec>", line 5, in get
  File "/opt/OpenvStorage/ovs/extensions/db/arakoon/pyrakoon/pyrakoon/client/utils.py", line 99, in wrapped
    return self._process(message) #pylint: disable=W0212
  File "/opt/OpenvStorage/ovs/extensions/db/arakoon/pyrakoon/pyrakoon/compat.py", line 1103, in _process
    connection.read)
  File "/opt/OpenvStorage/ovs/extensions/db/arakoon/pyrakoon/pyrakoon/utils.py", line 342, in read_blocking
    value = read_fun(request.count)
  File "/opt/OpenvStorage/ovs/extensions/db/arakoon/pyrakoon/pyrakoon/compat.py", line 1347, in read
    raise ArakoonSockRecvError
ArakoonSockRecvError: Error while receiving data from socket
```

**Comment by khenderick** [comment]
So apparently the client used for configuration wasn't cleared when a process was forked, so the socket was used by multiple processes at the same time, causing various issues

**Comment by khenderick** [comment]
Fixed by #943, packaged in openvstorage-2.7.3-rev.4013.c4ac0a4

**Comment by kinvaris** [comment]
PASSED: We've not seen the error in CI tests since the bug report
